crates/core_arch/src/acle/hints.rs

```diff
@@ -9,7 +9,8 @@
 /// low-power state until one of a number of asynchronous events occurs.
 // Section 10.1 of ACLE says that the supported arches are: 8, 6K, 6-M
 // LLVM says "instruction requires: armv6k"
-#[cfg(any(target_feature = "v6", target_arch = "aarch64"))]
+#[cfg(any(target_feature = "v6", target_arch = "aarch64", doc))]
+#[doc(cfg(target_arch = "arm"))]
```

Comment:
Adding `doc` to the cfg is correct: we want this function to be included in the documentation even if we aren't compiling for arm/aarch64 (e.g. the current target is x86).

However the `doc(cfg)` is not needed here. We don't need to mark this function as ARM-only (which is incorrect: this function is available on both ARM and AArch64). We already have a blanket `doc(cfg)` over the entire `arm` and `aarch64` top level modules.

Reply:
Am I correct in understanding that `aarch64` and `arm` will each re-export all of `acle`, so all the available intrinsics for a given target will get rendered together?

I'm not sure if there is known rationale for this, so please excuse my ignorance, but the naming of `acle` is a little awkward; it usually stands for "Arm C Language Extension", which would be fine if it were simple bindings, but it looks like a subset. Does a name like `arm_shared` make more sense here?

Out of curiosity, why does the same approach not work for `v6` and the like? When you say that they "aren't in the ARM spec", which specification are you referring to? There are definitely intrinsics available at `v7`, but not `v6`, for example. One argument for keeping them all is so that run-time feature detection (`std::detect`) can be used to guard intrinsics not available in `target_feature`.

Reply:
Yes, `arm_shared` would make more sense here. Let's rename `acle` to that.

I actually misread the ACLE spec, the `v6` and `v7` modules can stay in `arm`.

Reply:
Great, thanks for the hint, I found the blanket `doc(cfg)` and hope to slowly get the idea :). Should be fixed with https://github.com/rust-lang/stdarch/pull/1104/commits/8c6fdc5a1e33c9ca18cc96703296223ad0758f78 .